clientv3/concurrency/session.go

```diff
@@ -41,15 +41,20 @@ type Session struct {
 	donec  <-chan struct{}
 }
 
-// NewSession gets the leased session for a client.
-func NewSession(client *v3.Client) (*Session, error) {
+// NewSession gets the leased session for a client with given TTL in seconds.
+// If TTL is <= 0, the default 60 seconds TTL will be used.
+func NewSession(client *v3.Client, ttl int) (*Session, error) {
 	clientSessions.mu.Lock()
 	defer clientSessions.mu.Unlock()
 	if s, ok := clientSessions.sessions[client]; ok {
```

Comment:
@nekto0n the intent was to have one session per client which is typical in similar systems (e.g., chubby, zk) since it's only used as a mechanism to ensure the client is still live. The current model breaks down when requesting different TTLs on the same session. It needs some more thought.